hstream/src/HStream/Server/Core/Stream.hs

```diff
@@ -118,3 +116,11 @@ appendStream ServerContext{..} API.AppendRequest{..} partitionKey = do
   S.AppendCompletion {..} <- S.appendBatchBS scLDClient logId (V.toList payloads) cmpStrategy Nothing
   let records = V.zipWith (\_ idx -> API.RecordId appendCompLSN idx) appendRequestRecords [0..]
   return $ API.AppendResponse appendRequestStreamName records
+
+--------------------------------------------------------------------------------
+
+checkStreamExist :: CBytes -> ZHandle -> S.StreamId -> S.LDClient -> IO (Bool, Bool)
+checkStreamExist sCB zk sID ld = do
+  zkPathExist <- isJust <$> zooExists zk (streamRootPath <> "/" <> sCB)
+  streamExist <- S.doesStreamExist ld sID
+  return (zkPathExist, streamExist)
```

Comment:
Instead of checking and making, you can create a stream directly and catch the exceptions.